neo/Consensus/ConsensusService.cs

```diff
@@ -277,262 +364,39 @@ private void OnPersistCompleted(Block block)
-            {
-                for (ushort i = 0; i < context.Validators.Length; i++)
-                {
-                    if (i == tempContext.PrimaryIndex) continue;
-                    if (!message.PreparationMessages.ContainsKey(i)) continue;
-                    Log($" considering prepare request {i}");
-                    var regeneratedPrepareResponse = tempContext.RegenerateSignedPayload(prepareResponseMsg, i,
-                        message.PreparationMessages[i].InvocationScript);
-                    if (!regeneratedPrepareResponse.Verify(context.Snapshot) ||
-                        !PerformBasicConsensusPayloadPreChecks(regeneratedPrepareResponse)) continue;
-                    prepareResponses.Add((regeneratedPrepareResponse, prepareResponseMsg));
-                    Log($" verified prepare {i}");
-                    // Verify that there are M valid preparations, 1 Prepare Request + (M-1) Prepare responses
-                    if (prepareRequestMessage == null || prepareResponses.Count < context.M - 1) continue;
-                    canRestoreView = true;
-                    break;
-                }
-            }
-
-            var verifiedChangeViewPayloads = new ConsensusPayload[context.Validators.Length];
-            if (!canRestoreView && message.ChangeViewMessages.Count >= context.M)
-            {
-                var changeViewMsg = new ChangeView {NewViewNumber = message.ViewNumber};
-                int validChangeViewCount = 0;
-                for (ushort i = 0; i < context.Validators.Length; i++)
-                {
-                    if (!message.ChangeViewMessages.ContainsKey(i)) continue;
-
-                    changeViewMsg.ViewNumber = message.ChangeViewMessages[i].OriginalViewNumber;
-                    changeViewMsg.Timestamp = message.ChangeViewMessages[i].Timestamp;
-                    // Regenerate the ChangeView message
-                    var regeneratedChangeView = tempContext.RegenerateSignedPayload(changeViewMsg, i,
-                        message.ChangeViewMessages[i].InvocationScript);
-                    if (!regeneratedChangeView.Verify(context.Snapshot) ||
-                        !PerformBasicConsensusPayloadPreChecks(regeneratedChangeView)) continue;
-                    verifiedChangeViewPayloads[i] = regeneratedChangeView;
-                    validChangeViewCount++;
-                    if (validChangeViewCount < context.M) continue;
-                    canRestoreView = true;
-                    break;
-                }
-            }
-
-            if (!canRestoreView) return;
-            // We had enough valid change view messages or preparations to safely change the view number.
-            Log($"regenerating view: {message.ViewNumber}");
-
-            if (tempContext.PrimaryIndex == context.MyIndex && prepareRequestPayload != null)
-            {
-                // If we are the primary and there was a valid prepare request payload, we will accept our own prepare
```

Comment:
Does the code no longer accept it's own previously sent prepare request?

Reply:
I see that it does still support it. You just changed it so that the normal processing of a `PrepareRequest` can be accepted by it's own index. Won't this cause being able to accept it's own `PrepareRequest` and reprocess it even when receiving it's own messages not in the recovery state though?

Reply:
Actually, I see it is saved because it checks if either flag for `RequestSent` or `RequestReceived` is set. I'll resolve this conversation.